### PR TITLE
HPCC-26594 React ECL Watch pages not refreshing data

### PR DESCRIPTION
**Pull request by GordonSmith**
Refreshing the Table doesn't always refresh the data underneath.
Removed superfluous overflowButtonProps={{}}.
Added missing Refresh Buttons.
Split out Summary tabs from Files and Workunits.

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/src-react/hooks/file.ts [hunk 1]
```diff
@@ -1,40 +1,85 @@
...
+    return [defFile, increment];
+}
+
+export function useFileHistory(cluster: string, name: string): [WsDfu.Origin2[], () => void, () => void] {
+
+    const [file] = useFile(cluster, name);
+    const [history, setHistory] = React.useState<WsDfu.Origin2[]>([]);
+    const [count, increment] = useCounter();
+
+    const ereaseHistory = React.useCallback(() => {
```
Comment (jeclrsg) [inline esp/src/src-react/hooks/file.ts:64]: typo in var name? and on L84 in deps array

File: esp/src/src-react/util/throttle.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { debounce } from "@hpcc-js/util";
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+type TypeOfClassMethod<T, M extends keyof T> = T[M] extends Function ? T[M] : never;
+
+export function singletonDebounce<T, M extends keyof T>(obj: T, method: M): TypeOfClassMethod<T, M> {
+    const __lazy__ = Symbol.for(`__lazy__${method}`);
+    if (!obj[__lazy__]) {
+        obj[__lazy__] = debounce((...args: any[]) => (obj[method] as any)(...args), 1000);
```
Comment (jeclrsg) [inline esp/src/src-react/util/throttle.ts:9]: Would there be any usefulness in exposing the wait time?

File: esp/src/src/Memory.ts [hunk 3]
```diff
@@ -10,26 +16,51 @@ export interface MemoryOptions {
...
     getIdentity(object) {
         return object[this.idProperty];
     }
 
+    protected fetchData(): FetchDataResponse {
+        return Promise.resolve([]);
+    }
+
+    query(query, options) {
+        const retVal = new Deferred();
```
Comment (jeclrsg) [inline esp/src/src/Memory.ts:40]: Does the dojo Deferred behave differently than plain JS Promises?

**Review by jeclrsg — APPROVED** [review]

File: esp/src/src-react/util/throttle.ts (hunk 2 above)
Comment (GordonSmith) [reply esp/src/src-react/util/throttle.ts:9]: I wasn't sure - my thought was that it should be configurable and at least this way its in one place.

FYI - this is an "early" debounce.   IOW it lets the first call through and all the matching ones within a 1000ms get the same result.  You might be more familiar with a "late" debounce for things like resizing a window etc.

File: esp/src/src-react/hooks/file.ts (hunk 1 above)
Comment (GordonSmith) [reply esp/src/src-react/hooks/file.ts]: Fixed.

File: esp/src/src-react/util/throttle.ts (hunk 2 above)
Comment (jeclrsg) [reply esp/src/src-react/util/throttle.ts:9]: Ah, ok. Yes, that is what I was thinking of.

**Pull request merged**